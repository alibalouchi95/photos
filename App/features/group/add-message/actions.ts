import { createAsyncAction } from 'typesafe-actions'

export interface addMessagePayload {
  id: string
  groupId: string
}

export interface addMessageRequestPayload extends addMessagePayload {
  body: string
}

export interface addMessageFailurePayload extends addMessagePayload {
  error: any
}

export const addMessage = createAsyncAction(
  'group/add-message/ADD_MESSAGE_REQUEST',
  'group/add-message/ADD_MESSAGE_SUCCESS',
  'group/add-message/ADD_MESSAGE_FAILURE'
)<addMessageRequestPayload, addMessagePayload, addMessageFailurePayload>()
