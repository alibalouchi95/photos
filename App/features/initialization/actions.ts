import { createAction } from 'typesafe-actions'
import { InitializationStatus, InitializationPath } from './models'

export const initializeNewAccount = createAction(
  'initialization/INITIALIZE_CREATING_NEW_WALLET_AND_ACCOUNT'
)

export const initializeExistingAccount = createAction(
  'initialization/INITIALIZE_WITH_EXISTING_ACCOUNT_SEED',
  resolve => {
    return (seed: string) => resolve({ seed })
  }
)

export const updateInitializationStatus = createAction(
  'initialization/UPDATE_INITIALIZATION_STATUS',
  resolve => {
    return (status: InitializationStatus) => resolve({ status })
  }
)

export const failedToInitializeNode = createAction(
  'initialization/FAILED_TO_INITIALIZE_NODE',
  resolve => {
    return (error: any) => resolve({ error })
  }
)

export const chooseInitializationPath = createAction(
  'initialization/CHOOSE_INITIALIZATION_PATH',
  resolve => {
    return (path?: InitializationPath) => resolve({ path })
  }
)
