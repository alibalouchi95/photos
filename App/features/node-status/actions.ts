import { createAction } from 'typesafe-actions'

export const failedToInitializeNode = createAction(
  'node-status/FAILED_TO_INITIALIZE_NODE',
  resolve => (error: any) => resolve({ error })
)

export const nodeStarted = createAction('node-status/NODE_STARTED')

export const nodeStopped = createAction('node-status/NODE_STOPPED')

export const nodeOnline = createAction('node-status/NODE_ONLINE')

export const nodeFailedToStart = createAction(
  'node-status/NODE_FAILED_TO_START',
  resolve => (error: string) => resolve({ error })
)

export const nodeFailedToStop = createAction(
  'node-status/NODE_FAILED_TO_STOP',
  resolve => (error: string) => resolve({ error })
)

export const stopNodeAfterDelayStarting = createAction(
  'node-status/STOP_NODE_AFTER_DELAY_STARTING',
  resolve => (delay: number) => resolve({ delay })
)

export const stopNodeAfterDelayCancelled = createAction(
  'node-status/STOP_NODE_AFTER_DELAY_CANCELLED'
)
