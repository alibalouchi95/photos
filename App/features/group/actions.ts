import { feedActions as feed } from './feed'
import { addMessageActions as addMessage } from './add-message'
import { addPhotoActions as addPhoto } from './add-photo'
import { renameGroupActions as renameGroup } from './rename-group'
import { ignoreActions as ignore } from './ignore'

export { feed, addMessage, addPhoto, renameGroup, ignore }
