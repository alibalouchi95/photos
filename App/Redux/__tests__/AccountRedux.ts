import actions, { reducer } from '../AccountRedux'

const initialState = reducer(undefined, {} as any)
const info = {
  address: 'address',
  peerId: 'peerId',
  profile: {
    address: 'address',
    inboxes: ['address1', 'address2'],
    username: 'username',
    avatar_uri: 'avatarUri'
  },
  seed: 'seed',
  username: 'username'
}
const error = 'error'
const avatarId = 'avatarId'
const recoveryPhrase = 'recoveryPhrase'

describe('account', () => {
  describe('initial state', () => {
    it('should match snapshot', () => {
      expect(initialState).toMatchSnapshot()
    })
  })
  describe('account info', () => {
    it('should not change state for request', () => {
      const state0 =  reducer(initialState, actions.refreshAccountInfoRequest())
      expect(state0).toEqual(initialState)
    })
    it('should update account info', () => {
      const { address, peerId, profile, seed, username } = info
      const state0 = reducer(initialState, actions.refreshAccountInfoSuccess(address, peerId, profile, seed, username))
      expect(state0.info).toEqual(info)
    })
    it('should track account info error', () => {
      const state0 = reducer(initialState, actions.refreshAccountInfoError(error))
      expect(state0.info.error).toEqual(error)
    })
  })
  describe('avatar', () => {
    it('should not change state for request', () => {
      const state0 = reducer(initialState, actions.setAvatarRequest('avatarId'))
      expect(state0).toEqual(initialState)
    })
    it('should track avatar error', () => {
      const state0 = reducer(initialState, actions.setAvatarError(error))
      expect(state0.avatar.error).toEqual(error)
    })
    it('should track pending avatar id', () => {
      const state0 = reducer(initialState, actions.setPendingAvatar(avatarId))
      expect(state0.avatar.pendingId).toEqual(avatarId)
    })
  })
  describe('recovery phrase', () => {
    it('should track revocery phrase', () => {
      const state0 = reducer(initialState, actions.setRecoveryPhrase(recoveryPhrase))
      expect(state0.recoveryPhrase).toEqual(recoveryPhrase)
    })
  })
})
