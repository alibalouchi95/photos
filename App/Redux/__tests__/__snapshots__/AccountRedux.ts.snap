// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`account initial state should match snapshot 1`] = `
Object {
  "avatar": Object {},
  "info": Object {},
}
`;
